.github/workflows/go.yml

```diff
@@ -5,37 +5,42 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
+  workflow_dispatch:
 
 jobs:
   build:
     name: Lint Test and Build
     runs-on: ubuntu-latest
     steps:
       - name:
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Pull tags and refs
         run: git fetch --prune --unshallow
       - name: Check if version is updated
         run: |
           export ExpectedVersion=$(git describe --tags --abbrev=0)
           export DefinedVersion=$(cat cmd/gemini/Version)
-          if [ "$ExpectedVersion" != "$DefinedVersion" ]; then 
+          if [ "$ExpectedVersion" != "$DefinedVersion" ]; then
             echo "Expect to have '$ExpectedVersion', but got '$DefinedVersion'"
             exit 1
           fi
       - name: Setup Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v5
         with:
-          go-version: 1.20.4
-
+          go-version: '1.23'
       - name: Linting
-        run: |
-          make check
-
+        uses: reviewdog/action-golangci-lint@v2
```

Comment:
It is better to stick to `make check`, to make sure everything works the same on cicd and locally.